docs/data/data-grid/ai-assistant/ai-assistant.md

```diff
@@ -0,0 +1,162 @@
+MUI [offers this service](/x/react-data-grid/ai-assistant/#with-muis-service) as a part of a premium package add-on.
+Email us at [sales@mui.com](mailto:sales@mui.com) for more information.
+:::
+
+The AI assistant feature lets users interact with the Data Grid component using natural language.
+Type a prompt like "sort by name" or "show amounts larger than 1000" in the prompt input field and the Data Grid will update accordingly.
+In supported browsers, users can also prompt the assistant using their voice.
+
+To enable client-side of this feature, pass `enableAiAssistant` prop.
+
+
+To increase the accuracy of the language processing, provide example values for the available columns.
+This can be done in the following ways.
+
+:::info
+AI assistant demos use a utility function `mockPromptResolver()` to simulate the API that resolves user's prompts.
+In a real-world scenario, replace this with [MUI's](/x/react-data-grid/ai-assistant/#with-muis-service) or [your own](/x/react-data-grid/ai-assistant/#with-custom-service) processing service.
+
+`mockPromptResolver()` can handle a predefined set of prompts:
+
+- `sort by name`
+- `sort by company name and employee name`
+- `show people from the EU`
+- `order companies by amount of people`
+
+You can use suggestions to quickly enter prompts that are supported by the mock resolver.
+
+:::
+
+## Custom examples
+
+Use the `examples` prop in the `columns` array to provide custom examples as context for prompt processing.
+The `examples` prop should contain an array of possible values for its respective column.
+
+{{"demo": "AssistantWithExamples.js", "bg": "inline"}}
+
+## Use row data for examples
+
+Pass the `allowAiAssistantDataSampling` prop to use row data to generate examples.
+This is useful if you're dealing with non-sensitive data and want to skip creating custom examples for each column.
+
+Data is collected randomly at the cell level, which means that the examples for a given column might not come from the same rows.
+
+For the `aiAssistant.getPromptContext()` API method, pass the `allowDataSampling` flag as a parameter.
+
+```ts
+const context = React.useMemo(
+  () => apiRef.current.aiAssistant.getPromptContext(allowDataSampling),
+  [apiRef, allowDataSampling],
+);
+```
+
+{{"demo": "AssistantWithDataSampling.js", "bg": "inline"}}
+
+## Using Server-side data
+
+The example below shows how to combine prompt control with [server-side data](/x/react-data-grid/server-side-data/).
+
+{{"demo": "AssistantWithDataSource.js", "bg": "inline"}}
+
+## Processing service integration
+
+Natural language prompts must be processed by a service to understand what kinds of state changes must be applied to the Data Grid to match the user's request.
+You can use MUI's processing service or build your own.
+
+### With MUI's service
+
+The Data Grid provides all the necessary elements for integration with MUI's service.
+
+1. Enable the AI Assistant feature by adding the `enableAiAssistant` prop.
```

Comment:
I feel that from a user's pov, provisioning a backend comes earlier than integrating it with the client. Could we swap the order of points 1 and 2?

Update: Noticed Sam already commented a similar thing earlier 🙈